src/zenml/integrations/gcp/services/vertex_deployment.py

```diff
@@ -0,0 +1,431 @@
+    value = re.sub(r"[^a-z0-9\-_]", "-", value)
+    # Ensure it starts with a letter/number by prepending 'x' if needed
+    if not value[0].isalnum():
+        value = f"x{value}"
+    # Truncate to 63 chars to stay under limit
+    return value[:63]
+
+
+class VertexDeploymentConfig(VertexBaseConfig, ServiceConfig):
+    """Vertex AI service configurations."""
+
+    def get_vertex_deployment_labels(self) -> Dict[str, str]:
+        """Generate labels for the VertexAI deployment from the service configuration."""
+        labels = self.labels or {}
+        labels["managed_by"] = "zenml"
+        if self.pipeline_name:
+            labels["pipeline-name"] = sanitize_vertex_label(self.pipeline_name)
+        if self.pipeline_step_name:
+            labels["step-name"] = sanitize_vertex_label(
+                self.pipeline_step_name
+            )
+        if self.model_name:
+            labels["model-name"] = sanitize_vertex_label(self.model_name)
+        if self.service_name:
+            labels["service-name"] = sanitize_vertex_label(self.service_name)
+        return labels
+
+
+class VertexPredictionServiceEndpoint(BaseModel):
+    """Vertex AI Prediction Service Endpoint."""
+
+    endpoint_name: str
+    deployed_model_id: str
+    endpoint_url: Optional[str] = None
+
+
+class VertexServiceStatus(ServiceStatus):
+    """Vertex AI service status."""
+
+    endpoint: Optional[VertexPredictionServiceEndpoint] = None
+
+
+class VertexDeploymentService(BaseDeploymentService):
+    """Vertex AI model deployment service."""
+
+    SERVICE_TYPE = ServiceType(
+        name="vertex-deployment",
+        type="model-serving",
+        flavor="vertex",
+        description="Vertex AI inference endpoint prediction service",
+    )
+    config: VertexDeploymentConfig
+    status: VertexServiceStatus = Field(
+        default_factory=lambda: VertexServiceStatus()
+    )
+
+    def _initialize_gcp_clients(self) -> None:
+        """Initialize GCP clients with consistent credentials."""
+        # Initialize aiplatform with project and location
+        from zenml.integrations.gcp.model_deployers.vertex_model_deployer import (
+            VertexModelDeployer,
+        )
+
+        zenml_client = Client()
+        model_deployer = zenml_client.active_stack.model_deployer
+        if not isinstance(model_deployer, VertexModelDeployer):
+            raise RuntimeError(
+                "Active model deployer must be Vertex AI Model Deployer"
+            )
+
+        # get credentials from model deployer
+        credentials, project_id = model_deployer._get_authentication()
+
+        # Initialize aiplatform
+        aiplatform.init(
+            project=project_id,
+            location=self.config.location,
+            credentials=credentials,
+        )
+
+        # Initialize logging client
+        self.logging_client = vertex_logging.Client(
+            project=project_id, credentials=credentials
+        )
```

Comment:
This doesn't work for me, given that this is a Pydantic model and you can't simply set new attributes that aren't declared in the class definition:

```
ValidationError: 1 validation error for VertexDeploymentService
logging_client
  Object has no attribute 'logging_client' [type=no_such_attribute, input_value=<google.cloud.logging_v2....bject at 0x79c591f5dbd0>, input_type=Client]
    For further information visit https://errors.pydantic.dev/2.8/v/no_such_attribute
```